addons/mail/models/mail_thread.py

```diff
@@ -2357,6 +2357,24 @@ def _message_auto_subscribe_notify(self, partner_ids, template):
                 model_description=model_description,
             )
 
+    @api.multi
+    def _message_auto_subscribe_check_access(self, pid):
+        """ Check if the specified partner has read access rights on the records to autofollow it.
+        """
+        partner_id = self.env['res.partner'].sudo().browse(pid).exists()
+        # a partner with no users cannot autofollow
+        if not partner_id.user_ids:
+            return False
+
+        for user in partner_id.user_ids:
+            sudoed_records = self.sudo(user.id)
+            try:
+                sudoed_records.check_access_rights('read')
+                sudoed_records.check_access_rule('read')
+            except exceptions.AccessError:
+                return False
+        return True
```

Comment:
Shouldn't we reverse the return logic:
- `return True` after checking both accesses, because as soon as there is 1 user with access it's enough
- just `pass` in the except: block, to allow testing next users for that partner
- `return False` at the end of the method, if no user worked